R/ResultFlagsDependent.R

```diff
@@ -399,163 +398,94 @@ DepthProfileData <- function(.data,
-          # coalesce WQX.Depth.TargetUnit columns
-          check.data$WQX.Depth.TargetUnit <- dplyr::coalesce(
-            check.data$WQX.Depth.TargetUnit.x,
-            check.data$WQX.Depth.TargetUnit.y
-          )
-          # remove extra columns
-          check.data <- dplyr::select(check.data, -c(
-            "WQX.Depth.TargetUnit.x",
-            "WQX.Depth.TargetUnit.y"
-          ))
-        }
-      }
-    }
-
-    if (("ActivityBottomDepthHeightMeasure" %in% fields) == TRUE) {
-      # proceed only if ActivityDepthHeightMeasure.MeasureUnitCode has values other than NA
-      if (sum(is.na(check.data$ActivityBottomDepthHeightMeasure.MeasureUnitCode)) > 0) {
-        # Join target unit and conversion factor from unit.ref to .data by ActivityDepthHeightMeasure.MeasureUnitCode
-        check.data <- merge(check.data, unit.ref[, c("Code", "WQX.Depth.TargetUnit", "Conversion.Factor")],
-          by.x = "ActivityBottomDepthHeightMeasure.MeasureUnitCode",
-          by.y = "Code",
-          all.x = TRUE
-        )
-        # rename new columns
-        check.data <- check.data %>%
-          dplyr::rename(WQX.ActBottomDepth.ConversionFactor = Conversion.Factor)
-        # check if Target Unit column already exists, if it does, combine columns
-        if (all(c("WQX.Depth.TargetUnit.x", "WQX.Depth.TargetUnit.y") %in% colnames(check.data)) == TRUE) {
-          # coalesce WQX.Depth.TargetUnit columns
-          check.data$WQX.Depth.TargetUnit <- dplyr::coalesce(
-            check.data$WQX.Depth.TargetUnit.x,
-            check.data$WQX.Depth.TargetUnit.y
+    # Columns to append
+    appCols <- c("WQX.ActDepth.ConversionFactor",
+                 "WQX.ActTopDepth.ConversionFactor",
+                 "WQX.ActBottomDepth.ConversionFactor",
+                 "WQX.ResultDepth.ConversionFactor",
+                 "WQX.Depth.TargetUnit")
+    for (i in seq_along(validFields)) {
+      field <- validFields[i]
+      if ((field %in% fields) == TRUE) {
+        # Old unit column
+        unitCol <- paste(field, ".MeasureUnitCode", sep="")
+        # New column to append
+        appCol <- appCols[i]
+        # proceed only if unitCol has values other than NA
+        if (sum(!is.na(check.data[unitCol])) > 0) {
+          targetUnit <- "WQX.Depth.TargetUnit"
+          # Join targetUnit and conversion factor from unit.ref to .data by unitCol
+          check.data <- merge(check.data, unit.ref[, c("Code", targetUnit, "Conversion.Factor")],
+                              by.x = unitCol,
+                              by.y = "Code",
+                              all.x = TRUE
           )
-          # remove extra columns
-          check.data <- dplyr::select(check.data, -c(
-            "WQX.Depth.TargetUnit.x",
-            "WQX.Depth.TargetUnit.y"
-          ))
+          # rename new columns
+          check.data <- check.data %>%
+            dplyr::rename(!!appCol := Conversion.Factor)
```

Comment:
@jbousquin Just an FYI:

Line 425 introduced a new note (below). I was able to remove it by adding **@importFrom rlang :=** to the documentation for DepthProfileData function. & Then added rlang as import in description file.

❯ checking R code for possible problems ... NOTE
  DepthProfileData: no visible global function definition for ':='
  Undefined global functions or variables:
    :=